### PR TITLE
ci: Bump NodeJS versions and OS to Ubuntu 22.04

### PR DESCRIPTION
**Pull request by ypid**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,17 +10,17 @@ name: Continuous Integration
 
 jobs:
   build:
-    runs-on: "ubuntu-20.04"
+    runs-on: "ubuntu-22.04"
```
Comment (KristjanESPERANTO) [inline .github/workflows/ci.yml:13]: Would it make more sense to use `ubuntu-latest`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -10,17 +10,17 @@ name: Continuous Integration
...
-          - "10.x"
-          - "12.x"
           - "14.x"
-          - "15.x"
+          - "16.x"
+          - "17.x"
+          - "18.x"
 
     steps:
       - uses: actions/checkout@v2
```
Comment (KristjanESPERANTO) [inline .github/workflows/ci.yml:26]: Shall we upgrade to "v3" here?

Comment (KristjanESPERANTO) [reply .github/workflows/ci.yml:26]: And also `setup-node@v3`?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ypid) [reply .github/workflows/ci.yml:13]: Good idea.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ypid) [reply .github/workflows/ci.yml:26]: Thanks for the hint!

**Pull request merged**